app/scripts/app.js

```diff
@@ -102,24 +102,41 @@ angular.module('yourStlCourts').config(function($stateProvider, $urlRouterProvid
       }
     })
     .state('paymentOptions', {
-      url: '/paymentOptions/:citationId',
-      templateUrl: 'views/PaymentOptions.html',
-      controller: 'PaymentOptionsCtrl as ctrl',
+      url: '/paymentOptions/{citation}/{dob:.*}',
+      templateUrl: 'views/citationInfo.html',
+      controller: 'CitationInfoCtrl as ctrl',
       resolve: {
-        citation: function($stateParams, Citations) {
-          return Citations.getByCitationId($stateParams.citationId);
+        citations: function($stateParams, Citations,Errors) {
+          if (!$stateParams.citation || !$stateParams.dob){
+            throw Errors.makeError(Errors.ERROR_CODE.BAD_REQUEST, "No tickets were found with the information provided.");
+          }else{
+            var params = {
+              dob: $stateParams.dob,
+              citationNumber:$stateParams.citation
+            };
+            return Citations.find(params).then(function(result){
+              if(result.citations.length <= 0) {
+                throw Errors.makeError(Errors.ERROR_CODE.BAD_REQUEST, "No tickets were found with the information provided.");
+              }
+              return result.citations;
+            });
+          }
+        },
+        faqData: function($http) {
+          return $http.get('data/questionAnswers.json').then(function(data){
```

Comment:
Now that you're making this call in more than 1 spot, it's time to bury this in a service method (that just returns data.data from within so you always get the objects you want from everywhere that calls it